extensions/amp-story/amp-story-ads.md

```diff
@@ -0,0 +1,178 @@
+single page ads have a predefined set of call to action buttons and those call to action buttons take a web landing page URL where the user is navigated to, on click.
+
+![Story Page Ad](img/story-page-ad.png)
+
+### Sponsored story ad 
+When publishing sponsored story ads, use the `<amp-story ad>` tag instead of `<amp-story>`, so platforms can surface these stories appropriately. Since a sponsored story ad is just a URL on the web, you can also send user traffic to a sponsored story ad from the CTA URL button of a single page ad. 
+
+![Sponsored Story Ad](img/sponsored-story-ad.png)
+
+## Consistent UX
+
+The ad label and the call to action (CTA) buttons on the ads must be consistent across all publishers and ad networks. Therefore, the AMP stories runtime takes responsibility of rendering those in a consistent manner. 
+
+![Consistent Ad UX](img/consistent-ux.png)
+
+
+## Publisher placed ads 
+This is an option for publishers who would like to place single page ads amidst AMP story content they produce. This is meant as a stop-gap until more robust ad server support is available. 
+
+This is done using the mechanism in [Custom ad](../../ads/custom.md) extension. The ads are rendered with inlined templates in the story document
+and the data for the templates is fetched remotely.
+
+### Template
+An ad template must be written in [amp-mustache](../amp-mustache/amp-mustache.md).
+For example:
+
+```html
+<template type="amp-mustache" id="template-1">
+  <amp-img src="{{imgSrc}}"></amp-img>
+  <amp-pixel src="{{impressionUrl}}"></amp-pixel>
+</template>
+```
+
+A few important things to note: 
+
+* Templates need to be inlined in the AMP story, as direct children of a `<amp-story-auto-ads>` element.
+* An element ID is required, so that the template can be referenced by the ad response.
+* The selected template ID will be set as an attribute of `amp-ad`: `<amp-ad template="template-1">`
+* The content inside a template should strictly follow the [rules](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/validator-amp-story.protoascii) of `amp-story-grid-layer`
+* Be aware of the [restrictions](../amp-mustache/amp-mustache.md#restrictions) of `amp-mustache`.
+* Ads that use different templates can be styled separately using CSS attribute selector:
+```css
+amp-ad[template=template-1] {
+  background-color: blue;
+}
+amp-ad[template=template-2] {
+  background-color: red;
+}
+```
+- The CTA (call-to-action) button should NOT
+be included in the template. Story defines a list of CTA buttons to select from.
+For details, read the ["CTA ad" section](#cta-ad) below. 
+
+### Ad response
+
+#### Response payload
+A server endpoint needs to provide ad responses in the following JSON format:
+
+```json
+{
+  "templateId": "template-1",
+  "data": {
+    "imgSrc": "https://cdn.adserver.com/img-12345.jpg",
+    "impressionUrl": "https://adserver.com/track?iid=18745543"
+  },
+  "var": {
```

Comment:
@lannka & I just discovered this should actually be `vars` to work correctly with `amp-ad` 😬